lib/touch.js

```diff
@@ -129,58 +129,57 @@ const Injector = ObjC.registerClass({
 
         const app = UIApplication.sharedApplication();
         const event = app['- _touchesEvent'].call(app);
-        event['- _clearTouches'].call(event);
-
-        const absoluteTime = mach_absolute_time();
-
-        const timestamp = [
-            absoluteTime.shr(32).toNumber(),
-            absoluteTime.and(0xffffffff).toNumber()
-        ];
-
-        touch.setTimestamp_(CFGetSystemUptime());
-
-        const eventMask = (phase === UITouchPhaseMoved)
-            ? kIOHIDDigitizerEventPosition
-            : (kIOHIDDigitizerEventRange | kIOHIDDigitizerEventTouch);
-        const isRangeAndTouch = (phase !== UITouchPhaseEnded) ? 1 : 0;
-
-        const hidEvent = IOHIDEventCreateDigitizerFingerEvent(
-            kCFAllocatorDefault,
-            timestamp,
-            0,
-            2,
-            eventMask,
-            point[0],
-            point[1],
-            0,
-            0,
-            0,
-            isRangeAndTouch,
-            isRangeAndTouch,
-            0);
-
-        if ('- _setHidEvent:' in touch) {
-          touch['- _setHidEvent:'].call(touch, hidEvent);
-        }
-
-        event['- _setHIDEvent:'].call(event, hidEvent);
-        event['- _addTouch:forDelayedDelivery:'].call(event, touch, false);
-
-        const pool = NSAutoreleasePool.alloc().init();
-        try {
-          app.sendEvent_(event);
-        } finally {
-          pool.release();
+        if (event !== null) {
```

Comment:
Oh wait, sorry, I'm not sure I understand what happens here. Does `-[UIApplication _touchesEvent]` keep returning `nil` or is it a random glitch? If it's the latter I'm wondering if we should make the `_touchesEvent()` call earlier, before we do `pending.shift()`, and simply return there -- that way `dispatchTouch:` gets called again on the next frame, and might succeed that time and keep going.

Reply:
It occurs when the application goes to the background for any reason and comes back.  So I would say it's happening on a non-predictable manner.

Reply:
Interesting. In that case I think we have two options, and I'm not yet sure which one is the most correct:
1) Keep trying, by returning early, before we do `pending.shift()`. In this way we will eventually move it through the different phases.
2) Complete the touch right away, as if `pending.length === 0`. In that case we will have to release `priv.touch` if we already created it.

What do you think, @mrmacete?